presto-delta/src/test/java/com/facebook/presto/delta/TestDeltaOptimizationsTests.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.delta;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.predicate.Domain;
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.cost.StatsProvider;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.sql.planner.assertions.MatchResult;
+import com.facebook.presto.sql.planner.assertions.Matcher;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.assertions.SymbolAliases;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static com.facebook.presto.common.predicate.Domain.multipleValues;
+import static com.facebook.presto.common.predicate.Domain.notNull;
+import static com.facebook.presto.common.predicate.Domain.onlyNull;
+import static com.facebook.presto.common.predicate.Domain.singleValue;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.sql.planner.assertions.MatchResult.NO_MATCH;
+import static com.facebook.presto.sql.planner.assertions.MatchResult.match;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.lang.String.format;
+
+/**
+ * Integrations tests for various optimization (such as filter pushdown, nested column project/filter pushdown etc)
+ * that speed up reading data from Delta tables.
+ */
+public class TestDeltaOptimizationsTests
+        extends AbstractDeltaDistributedQueryTestBase
+{
+    @Test
+    public void filterOnRegularColumn()
+    {
+        String tableName = "data-reader-primitives";
```

Comment:
Can you add the table definition as a comment for all these tests?

Reply:
Sorry, not clear. Are you asking for the table schema or data in the table?

Reply:
The table schema